### DISCUSSION TITLE
.

### DISCUSSION THREAD

**Discussion by ghost** [strongswan/strongswan#2113] [Q&A] [answered]

**Comment by tobiasbrunner** [comment]
Something doesn't match up here:

> keyexchange=ikev2

> Feb 21 17:57:44 VPN-nantes charon: 12[IKE] initiating Main Mode IKE_SA devgateway-to-prodgateway[1] to 172.16.3.29

Main Mode is IKEv1, to which the other peer doesn't respond, maybe it actually expects IKEv2. So I suggest you double check your config.

**Reply by Thermi** [reply]
> after all this i was thinking that it could work. but in the logs i have this : "/etc/syslog":

>Feb 21 17:57:44 VPN-nantes charon: 12[IKE] initiating Main Mode IKE_SA devgateway-to-prodgateway[1] to 172.16.3.29
> Feb 21 17:57:44 VPN-nantes charon: 12[ENC] generating ID_PROT request 0 [ SA V V V V V ]
> Feb 21 17:57:44 VPN-nantes charon: 12[NET] sending packet: from 172.16.3.26[500] to 172.16.3.29[500] (168 bytes)

> Listening IP addresses:
> 172.18.47.37

172.16.3.26 != 172.18.47.37

**Comment by Thermi** [comment, marked as answer]
The address you set in "left" has to be bound to a local interface 

On February 22, 2024 9:18:30 AM UTC, Sanktsu ***@***.***> wrote:
>yes sorry,
>i did i did a little mistake , i did show you the "statusall" of one of the vpn and the configuration of the other ^^'.
>
>just to clarify :
>london = Nantes
>Oxford = Marseille
>here's the configuration of the two vpn and their status :
>![help 3](https://github.com/strongswan/strongswan/assets/160762750/16249bfe-3114-4dd7-96c1-c4647d1ce809)
>
>![help 4](https://github.com/strongswan/strongswan/assets/160762750/88d445dd-66b2-43a5-ba6d-9ae4dca55260)
>
>
>and here's the log of London(Nantes):
>
>Feb 22 10:12:49 VPN-nantes charon: 13[IKE] initiating IKE_SA devgateway-to-prodgateway[1] to 172.16.3.29
>Feb 22 10:12:49 VPN-nantes charon: 13[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
>Feb 22 10:12:49 VPN-nantes charon: 13[NET] sending packet: from 172.16.3.26[500] to 172.16.3.29[500] (328 bytes)
>Feb 22 10:12:49 VPN-nantes charon: 16[NET] error writing to socket: Network is unreachable
>Feb 22 10:12:53 VPN-nantes charon: 12[IKE] retransmit 1 of request with message ID 0
>Feb 22 10:12:53 VPN-nantes charon: 12[NET] sending packet: from 172.16.3.26[500] to 172.16.3.29[500] (328 bytes)
>Feb 22 10:12:53 VPN-nantes charon: 16[NET] error writing to socket: Network is unreachable
>Feb 22 10:13:00 VPN-nantes charon: 14[IKE] retransmit 2 of request with message ID 0
>Feb 22 10:13:00 VPN-nantes charon: 14[NET] sending packet: from 172.16.3.26[500] to 172.16.3.29[500] (328 bytes)
>Feb 22 10:13:00 VPN-nantes charon: 16[NET] error writing to socket: Network is unreachable
>Feb 22 10:13:13 VPN-nantes charon: 15[IKE] retransmit 3 of request with message ID 0
>Feb 22 10:13:13 VPN-nantes charon: 15[NET] sending packet: from 172.16.3.26[500] to 172.16.3.29[500] (328 bytes)
>Feb 22 10:13:13 VPN-nantes charon: 16[NET] error writing to socket: Network is unreachable
>Feb 22 10:13:37 VPN-nantes charon: 04[IKE] retransmit 4 of request with message ID 0
>Feb 22 10:13:37 VPN-nantes charon: 04[NET] sending packet: from 172.16.3.26[500] to 172.16.3.29[500] (328 bytes)
>Feb 22 10:13:37 VPN-nantes charon: 16[NET] error writing to socket: Network is unreachable
>Feb 22 10:14:19 VPN-nantes charon: 05[IKE] retransmit 5 of request with message ID 0
>Feb 22 10:14:19 VPN-nantes charon: 05[NET] sending packet: from 172.16.3.26[500] to 172.16.3.29[500] (328 bytes)
>Feb 22 10:14:19 VPN-nantes charon: 16[NET] error writing to socket: Network is unreachable
>Feb 22 10:15:34 VPN-nantes charon: 09[IKE] giving up after 5 retransmits
>Feb 22 10:15:34 VPN-nantes charon: 09[IKE] peer not responding, trying again (5/0)
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/strongswan/strongswan/discussions/2113#discussioncomment-8554150
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
-- 
Sent from mobile. Please excuse any typos or brevity

**Comment by Thermi** [comment]
Check logs from the other side. It sent the error.

On February 22, 2024 9:43:56 AM UTC, Sanktsu ***@***.***> wrote:
>```
>No worry thank you for your response ^^.
>i've just changed it and i got a response ^^!
>```
>
>`Now i just need to resolve another problem ^^' do you have any idea ? `
>
>**here's the sys log**
>
>```
>Feb 22 10:38:08 VPN-nantes charon: 14[NET] received packet: from 172.16.3.29[500] to 172.18.47.37[500] (328 bytes)
>Feb 22 10:38:08 VPN-nantes charon: 14[ENC] parsed IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
>Feb 22 10:38:08 VPN-nantes charon: 14[IKE] no IKE config found for 172.18.47.37...172.16.3.29, sending NO_PROPOSAL_CHOSEN
>Feb 22 10:38:08 VPN-nantes charon: 14[ENC] generating IKE_SA_INIT response 0 [ N(NO_PROP) ]
>Feb 22 10:38:08 VPN-nantes charon: 14[NET] sending packet: from 172.18.47.37[500] to 172.16.3.29[500] (36 bytes)
>Feb 22 10:38:16 VPN-nantes charon: 00[DMN] SIGINT received, shutting down
>Feb 22 10:38:16 VPN-nantes charon: 00[IKE] destroying IKE_SA in state CONNECTING without notification
>Feb 22 10:38:18 VPN-nantes charon: 00[DMN] Starting IKE charon daemon (strongSwan 5.9.1, Linux 5.10.0-19-amd64, x86_64)
>Feb 22 10:38:18 VPN-nantes charon: 00[CFG] loading ca certificates from '/etc/ipsec.d/cacerts'
>Feb 22 10:38:18 VPN-nantes charon: 00[CFG] loading aa certificates from '/etc/ipsec.d/aacerts'
>Feb 22 10:38:18 VPN-nantes charon: 00[CFG] loading ocsp signer certificates from '/etc/ipsec.d/ocspcerts'
>Feb 22 10:38:18 VPN-nantes charon: 00[CFG] loading attribute certificates from '/etc/ipsec.d/acerts'
>Feb 22 10:38:18 VPN-nantes charon: 00[CFG] loading crls from '/etc/ipsec.d/crls'
>Feb 22 10:38:18 VPN-nantes charon: 00[CFG] loading secrets from '/etc/ipsec.secrets'
>Feb 22 10:38:18 VPN-nantes charon: 00[CFG]   loaded IKE secret for 172.16.3.26 172.16.3.29
>Feb 22 10:38:18 VPN-nantes charon: 00[LIB] loaded plugins: charon aesni aes rc2 sha2 sha1 md5 mgf1 random nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp dnskey sshkey pem openssl fips-prf gmp agent xcbc hmac gcm drbg attr kernel-netlink resolve socket-default connmark stroke updown eap-mschapv2 xauth-generic counters
>Feb 22 10:38:18 VPN-nantes charon: 00[LIB] dropped capabilities, running as uid 0, gid 0
>Feb 22 10:38:18 VPN-nantes charon: 00[JOB] spawning 16 worker threads
>Feb 22 10:38:18 VPN-nantes charon: 05[CFG] received stroke: add connection 'devgateway-to-prodgateway'
>Feb 22 10:38:18 VPN-nantes charon: 05[CFG] added configuration 'devgateway-to-prodgateway'
>Feb 22 10:38:18 VPN-nantes charon: 05[CFG] received stroke: initiate 'devgateway-to-prodgateway'
>Feb 22 10:38:18 VPN-nantes charon: 05[IKE] initiating IKE_SA devgateway-to-prodgateway[1] to 172.16.3.29
>Feb 22 10:38:18 VPN-nantes charon: 05[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
>Feb 22 10:38:18 VPN-nantes charon: 05[NET] sending packet: from 172.18.47.37[500] to 172.16.3.29[500] (328 bytes)
>Feb 22 10:38:18 VPN-nantes charon: 09[NET] received packet: from 172.16.3.29[500] to 172.18.47.37[500] (336 bytes)
>Feb 22 10:38:18 VPN-nantes charon: 09[ENC] parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(CHDLESS_SUP) N(MULT_AUTH) ]
>Feb 22 10:38:18 VPN-nantes charon: 09[CFG] selected proposal: IKE:AES_GCM_16_256/PRF_HMAC_SHA2_256/MODP_1024
>Feb 22 10:38:18 VPN-nantes charon: 09[IKE] local host is behind NAT, sending keep alives
>Feb 22 10:38:18 VPN-nantes charon: 09[IKE] remote host is behind NAT
>Feb 22 10:38:18 VPN-nantes charon: 09[IKE] authentication of '172.18.47.37' (myself) with pre-shared key
>Feb 22 10:38:18 VPN-nantes charon: 09[IKE] establishing CHILD_SA devgateway-to-prodgateway{1}
>Feb 22 10:38:18 VPN-nantes charon: 09[ENC] generating IKE_AUTH request 1 [ IDi N(INIT_CONTACT) IDr AUTH SA TSi TSr N(MOBIKE_SUP) N(NO_ADD_ADDR) N(MULT_AUTH) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
>Feb 22 10:38:18 VPN-nantes charon: 09[NET] sending packet: from 172.18.47.37[4500] to 172.16.3.29[4500] (253 bytes)
>Feb 22 10:38:18 VPN-nantes charon: 10[NET] received packet: from 172.16.3.29[4500] to 172.18.47.37[4500] (65 bytes)
>Feb 22 10:38:18 VPN-nantes charon: 10[ENC] parsed IKE_AUTH response 1 [ N(AUTH_FAILED) ]
>Feb 22 10:38:18 VPN-nantes charon: 10[IKE] received AUTHENTICATION_FAILED notify error
>```
>
>
>here's the ipsec.conf with modification:
>
>```
>config setup
>        charondebug="all"
>        uniqueids=yes
>conn devgateway-to-prodgateway
>        type=tunnel
>        auto=start
>        keyexchange=ikev2
>        authby=psk
>        left=172.18.47.37
>        leftsubnet=172.18.47.32/28
>        right=172.16.3.29
>        rightsubnet=172.18.49.1/28
>        ike=aes256gcm16-sha256-modp1024!
>        esp=aes256gcm16-sha256-modp1024!
>        keyingtries=%forever
>        ikelifetime=60s
>        lifetime=30s
>        dpddelay=10s
>        dpdtimeout=1200s
>        dpdaction=restart
>
>```
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/strongswan/strongswan/discussions/2113#discussioncomment-8554481
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
-- 
Sent from mobile. Please excuse any typos or brevity

**Comment by Thermi** [comment]
Verify IP addresses and IDs on the two sides match the configs 

On February 22, 2024 9:56:26 AM UTC, Sanktsu ***@***.***> wrote:
>it looks like the peer config is not the same :/
>
>do you have any idea from it ?
>```
>Feb 22 10:38:06 VPN-marseille charon: 00[DMN] SIGINT received, shutting down
>Feb 22 10:38:06 VPN-marseille charon: 00[IKE] destroying IKE_SA in state CONNECTING without notification
>Feb 22 10:38:08 VPN-marseille charon: 00[DMN] Starting IKE charon daemon (strongSwan 5.9.1, Linux 5.10.0-19-amd64, x86_64)
>Feb 22 10:38:08 VPN-marseille charon: 00[CFG] loading ca certificates from '/etc/ipsec.d/cacerts'
>Feb 22 10:38:08 VPN-marseille charon: 00[CFG] loading aa certificates from '/etc/ipsec.d/aacerts'
>Feb 22 10:38:08 VPN-marseille charon: 00[CFG] loading ocsp signer certificates from '/etc/ipsec.d/ocspcerts'
>Feb 22 10:38:08 VPN-marseille charon: 00[CFG] loading attribute certificates from '/etc/ipsec.d/acerts'
>Feb 22 10:38:08 VPN-marseille charon: 00[CFG] loading crls from '/etc/ipsec.d/crls'
>Feb 22 10:38:08 VPN-marseille charon: 00[CFG] loading secrets from '/etc/ipsec.secrets'
>Feb 22 10:38:08 VPN-marseille charon: 00[CFG]   loaded IKE secret for 172.16.3.29 172.16.3.26
>Feb 22 10:38:08 VPN-marseille charon: 00[LIB] loaded plugins: charon aesni aes rc2 sha2 sha1 md5 mgf1 random nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp dnskey sshkey pem openssl fips-prf gmp agent xcbc hmac gcm drbg attr kernel-netlink resolve socket-default connmark stroke updown eap-mschapv2 xauth-generic counters
>Feb 22 10:38:08 VPN-marseille charon: 00[LIB] dropped capabilities, running as uid 0, gid 0
>Feb 22 10:38:08 VPN-marseille charon: 00[JOB] spawning 16 worker threads
>Feb 22 10:38:08 VPN-marseille charon: 05[CFG] received stroke: add connection 'devgateway-to-prodgateway'
>Feb 22 10:38:08 VPN-marseille charon: 05[CFG] added configuration 'devgateway-to-prodgateway'
>Feb 22 10:38:08 VPN-marseille charon: 07[CFG] received stroke: initiate 'devgateway-to-prodgateway'
>Feb 22 10:38:08 VPN-marseille charon: 07[IKE] initiating IKE_SA devgateway-to-prodgateway[1] to 172.16.3.26
>Feb 22 10:38:08 VPN-marseille charon: 07[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
>Feb 22 10:38:08 VPN-marseille charon: 07[NET] sending packet: from 172.18.49.1[500] to 172.16.3.26[500] (328 bytes)
>Feb 22 10:38:09 VPN-marseille charon: 09[NET] received packet: from 172.16.3.26[500] to 172.18.49.1[500] (36 bytes)
>Feb 22 10:38:09 VPN-marseille charon: 09[ENC] parsed IKE_SA_INIT response 0 [ N(NO_PROP) ]
>Feb 22 10:38:09 VPN-marseille charon: 09[IKE] received NO_PROPOSAL_CHOSEN notify error
>Feb 22 10:38:18 VPN-marseille charon: 11[NET] received packet: from 172.16.3.26[500] to 172.18.49.1[500] (328 bytes)
>Feb 22 10:38:18 VPN-marseille charon: 11[ENC] parsed IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
>Feb 22 10:38:18 VPN-marseille charon: 11[IKE] 172.16.3.26 is initiating an IKE_SA
>Feb 22 10:38:18 VPN-marseille charon: 11[CFG] selected proposal: IKE:AES_GCM_16_256/PRF_HMAC_SHA2_256/MODP_1024
>Feb 22 10:38:18 VPN-marseille charon: 11[IKE] local host is behind NAT, sending keep alives
>Feb 22 10:38:18 VPN-marseille charon: 11[IKE] remote host is behind NAT
>Feb 22 10:38:18 VPN-marseille charon: 11[ENC] generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(CHDLESS_SUP) N(MULT_AUTH) ]
>Feb 22 10:38:18 VPN-marseille charon: 11[NET] sending packet: from 172.18.49.1[500] to 172.16.3.26[500] (336 bytes)
>Feb 22 10:38:18 VPN-marseille charon: 12[NET] received packet: from 172.16.3.26[4500] to 172.18.49.1[4500] (253 bytes)
>Feb 22 10:38:18 VPN-marseille charon: 12[ENC] parsed IKE_AUTH request 1 [ IDi N(INIT_CONTACT) IDr AUTH SA TSi TSr N(MOBIKE_SUP) N(NO_ADD_ADDR) N(MULT_AUTH) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
>**Feb 22 10:38:18 VPN-marseille charon: 12[CFG] looking for peer configs matching 172.18.49.1[172.16.3.29]...172.16.3.26[172.18.47.37]
>Feb 22 10:38:18 VPN-marseille charon: 12[CFG] no matching peer config found**
>Feb 22 10:38:18 VPN-marseille charon: 12[IKE] peer supports MOBIKE
>Feb 22 10:38:18 VPN-marseille charon: 12[ENC] generating IKE_AUTH response 1 [ N(AUTH_FAILED) ]
>Feb 22 10:38:18 VPN-marseille charon: 12[NET] sending packet: from 172.18.49.1[4500] to 172.16.3.26[4500] (65 bytes)
>```
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/strongswan/strongswan/discussions/2113#discussioncomment-8554650
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
-- 
Sent from mobile. Please excuse any typos or brevity

**Comment by Thermi** [comment]
Strongswan presumes the left and right values to be used as IDs if you use ask authentication. Set leftid and rightid correspondingly so they match.

On February 22, 2024 12:29:41 PM UTC, Sanktsu ***@***.***> wrote:
>> Verify IP addresses and IDs on the two sides match the configs
>
>**Thanks you for your response ,i have check the ip there is nothing that is wrong for me .
>the IP are not the same for the configuration because i need to go through the the public ip but it sould be okay ? 
>And i didn't add IDs(i think) where should i add it if i need it ?
>
>thank you for your response you save me ^^'.**
>
>
>**conf ipsec.conf of marseille(oxford):**
>
>```
>config setup
>        charondebug="all"
>        uniqueids=yes
>conn devgateway-to-prodgateway
>        type=tunnel
>        auto=start
>        keyexchange=ikev2
>        authby=psk
>        left=172.18.49.1
>        leftsubnet=172.18.49.0/28
>        right=172.16.3.26
>        rightsubnet=172.18.47.32/28
>        ike=aes256gcm16-sha256-modp1024!
>        esp=aes256gcm16-sha256-modp1024!
>        keyingtries=%forever
>        ikelifetime=60s
>        lifetime=30s
>        dpddelay=10s
>        dpdtimeout=1200s
>        dpdaction=restart
>```
>
>**conf ipsec.conf of Nantes(london):**
>
>
>```
>
>config setup
>        charondebug="all"
>        uniqueids=yes
>conn devgateway-to-prodgateway
>        type=tunnel
>        auto=start
>        keyexchange=ikev2
>        authby=psk
>        left=172.18.47.37
>        leftsubnet=172.18.47.32/28
>        right=172.16.3.29
>        rightsubnet=172.18.49.0/28
>        ike=aes256gcm16-sha256-modp1024!
>        esp=aes256gcm16-sha256-modp1024!
>        keyingtries=%forever
>        ikelifetime=60s
>        lifetime=30s
>        dpddelay=10s
>        dpdtimeout=1200s
>        dpdaction=restart
>
>```
>
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/strongswan/strongswan/discussions/2113#discussioncomment-8556208
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
-- 
Sent from mobile. Please excuse any typos or brevity
